### ISSUE TITLE
Increase the width of TextCtrls used to enter a widget's class?

### ISSUE THREAD

**Issue by reticulatus** [wxGlade/wxGlade#541]
When adding a Custom Widget to a design, an "Enter widget class" dialog appears. 

That dialog has a small TextCtrl in which to enter the Instance class. On my linux PC it only has space for 10 characters before it starts to scroll horizontally.

The names that I use for custom widgets' classes are usually much larger than this, which means I then need to use the left/right arrow keys to check I have typed it correctly.

I realise that I could just press OK button and then check/correct in the Properties window, but it would be nice to be able to more easily get it right first time.

The 'class' TextCtrls in the dialogs for adding a Frame or Dialog are also a bit short for the typical class names I use, although not as bad as the Custom Widget case.

Please could you consider making the TextCtrls in these dialogs wider (e.g. 30 characters)?

**Comment by DietmarSchwertberger** [comment]
I have implemented a modification in the master branch.
Please try it. Would you mind posting screenshots before and after?
On Windows I did not see a problem. I was setting the dialog width to 2x the width of the title and that was wide enough.

**Comment by reticulatus** [comment]
Thank you.

Here is using v1.0.5

![Screenshot at 2024-02-18 16-22-18](https://github.com/wxGlade/wxGlade/assets/6072789/6a531b3f-49a9-4760-8d9a-c8bf16abd290)

Here is using current master branch:

![Screenshot at 2024-02-18 16-26-58](https://github.com/wxGlade/wxGlade/assets/6072789/b349f476-342d-4a3d-8fef-c20134a3d2e4)

Much better!

For completeness here are the other dialogs on v1.0.5

![Screenshot at 2024-02-18 16-22-00](https://github.com/wxGlade/wxGlade/assets/6072789/00db1daf-1afc-4ad9-b804-aa833c86f649)

![Screenshot at 2024-02-18 16-22-48](https://github.com/wxGlade/wxGlade/assets/6072789/e2ffc976-132d-477e-a303-afa9bb08b27a)

I am using wxPython 4.2.1 + Python 3.10.12 + Linux Mint 21.3

**Comment by DietmarSchwertberger** [comment]
I don't fully understand the old behaviour. It was supposed to have at least the size of 2x the title text, but acc. to your screenshot it did not.
(`self.GetTextExtent(title)[0] *2`)

**Comment by reticulatus** [comment]
I've just checked what are the longest class names I have used for Frames and Dialogs in my current projects:

ScreenSaverPictureSelectorMainFrameBase
InventoryReportsResultsDialogBase

**Comment by reticulatus** [comment]
I think I see what has happened. My 'before' screen dumps were made using the current official release v1.0.5. 

That version does not appear to have the changes made by commit 8e78cdc.

In particular it has
```
        w = self.GetTextExtent(title)[0] + 50
```
and not
```
        w = self.GetTextExtent(title)[0] *2
```

I have now retested it with a copy of v1.1.0a2 that I had downloaded on 2023-11-30 which does include 8e78cdc and it does show a wider dialog:

![Screenshot at 2024-02-18 18-39-28](https://github.com/wxGlade/wxGlade/assets/6072789/807411f9-6348-40b5-adf8-64ea99874d47)

Sorry about the confusion, I should have checked the current master branch before raising this issue.
